File: src/main/java/com/timyr_tm/rust_bound/world/item/Items.java
package com.timyr_tm.rust_bound.world.item;

import com.timyr_tm.rust_bound.RustBound;
import com.timyr_tm.rust_bound.world.block.Blocks;
import com.timyr_tm.rust_bound.world.electricity.WireTypes;
import net.minecraft.world.item.BlockItem;
import net.neoforged.neoforge.registries.DeferredItem;
import net.neoforged.neoforge.registries.DeferredRegister;

public final class Items {
    public static final DeferredRegister.Items ITEMS = DeferredRegister.createItems(RustBound.MOD_ID);

    public static final DeferredItem<BlockItem> TEST_CONNECTOR = ITEMS.registerSimpleBlockItem(Blocks.TEST_CONNECTOR);
    public static final DeferredItem<WireCoilItem> TEST_COIL = ITEMS.registerItem("test_coil", properties -> new WireCoilItem(WireTypes.TEST_WIRE_TYPE.getKey(), properties));
}

File: src/main/kotlin/com/timyr_tm/rust_bound/world/item/Items.kt
package com.timyr_tm.rust_bound.world.item

import com.timyr_tm.rust_bound.RustBound
import com.timyr_tm.rust_bound.world.block.Blocks
import com.timyr_tm.rust_bound.world.WireTypes
import net.minecraft.world.item.BlockItem
import net.neoforged.neoforge.registries.DeferredItem
import net.neoforged.neoforge.registries.DeferredRegister

object Items {
    val ITEMS: DeferredRegister.Items = DeferredRegister.createItems(RustBound.MOD_ID)
    val TEST_SPOOL: DeferredItem<SpoolItem> = ITEMS.registerItem("test_spool") {
        SpoolItem(WireTypes.TEST_WIRE_TYPE, it)
    }
    val WIRE_SPOOL: DeferredItem<SpoolItem> = ITEMS.registerItem("copper_spool") {
        SpoolItem(WireTypes.COPPER_WIRE_TYPE, it)
    }

    val COPPER_INSULATOR: DeferredItem<BlockItem> = ITEMS.registerSimpleBlockItem(Blocks.COPPER_INSULATOR)

    val WIRE_CUTTER: DeferredItem<WireCutterItem> = ITEMS.registerItem("wire_cutter", ::WireCutterItem)
}
